cli_widgets/stdin_hide_guard.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <iostream>
+#include <boost/predef.h>
+#if BOOST_OS_WINDOWS
+#include <windows.h>
+#else
+#include <termios.h>
+#include <unistd.h>
+#endif
+
+namespace cli_widgets{
+  struct stdin_hide_guard {
+    stdin_hide_guard() {
+#if BOOST_OS_WINDOWS
+      HANDLE hStdin = GetStdHandle(STD_INPUT_HANDLE);
+      DWORD mode;
+      GetConsoleMode(hStdin, &mode);
+      mode &= ~ENABLE_ECHO_INPUT;
+      SetConsoleMode(hStdin, mode);
+#else
+      struct termios tty;
+      tcgetattr(STDIN_FILENO, &tty);
+      tty.c_lflag &= ~ECHO;
+      (void)tcsetattr(STDIN_FILENO, TCSANOW, &tty);
+#endif
+    }
+
+    ~stdin_hide_guard() {
+#if BOOST_OS_WINDOWS
+      HANDLE hStdin = GetStdHandle(STD_INPUT_HANDLE);
+      DWORD mode;
+      GetConsoleMode(hStdin, &mode);
+      mode |= ENABLE_ECHO_INPUT;
+      SetConsoleMode(hStdin, mode);
+#else
+      struct termios tty;
+      tcgetattr(STDIN_FILENO, &tty);
+      tty.c_lflag |= ECHO;
+      (void)tcsetattr(STDIN_FILENO, TCSANOW, &tty);
+#endif
```

Comment:
Well the implementation before with a central function that you could pass true and false  was cleanier. I mean you could call the function inside the RAII wrapper.

Reply:
It's just a matter of oring/xoring the ECHO flag, then the code can be divided by half in size.